### PR TITLE
Adding cryptoGen.sh documentation.

### PR DESCRIPTION
**Pull request by gennadylaventman**
Signed-off-by: Gennady Laventman <gennady@il.ibm.com>

### CODE REVIEW

File: docs/getting-started/launching-one-node/binary.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ For additional health check, we can run `make test` to ensure all tests pass.
 
 To start a node, we need a certificate authority and crypto materials for the node and admin users. To simplify this task, we have provided sample
 crypto materials and configuration files in `deployment/sample/`. In order to understand the configuration files and procedure to create crypto
-materials, refer to [TODO].
+materials, refer to [TODO] and [crypto-materials](crypto-materials).
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/binary.md:60]: why keep this [todo]? maybe a hidden comment with a reference to an issue?

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 2]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+   * `admin.certificatePath`
+
+To work with Orion server, each client needs:
+1. CA certificate.
+```go 
+&config.ConnectionConfig{     
+   RootCAs: []string{
+      "./crypto/CA/CA.pem",
+   },
+}
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:35]: do we support intermediate CAs in the client SDK?

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 3]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
  SPDX-License-Identifier: CC-BY-4.0
  -->
 
+### Crypto materials configuration
+
+Hyperledger Orion uses public key cryptography from user side, for user authentication and transaction Non-Repudiation. From server side, Orion uses PKI to sign and verify all server responses to clients.
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:14]: Hyperledger Orion uses public key cryptography to achieve authentication and transaction non-repudiation. This holds for both the client and the server. Therefore, each client signs the queries and transactions it submits to the server, and each server signs the responses it sends to the client.

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 4]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
  SPDX-License-Identifier: CC-BY-4.0
  -->
 
+### Crypto materials configuration
+
+Hyperledger Orion uses public key cryptography from user side, for user authentication and transaction Non-Repudiation. From server side, Orion uses PKI to sign and verify all server responses to clients. 
+
+Thus, each Orion client needs certificate/key pair to sign each query and transaction he submits to server, and server has to have its own certificate/key pair to sign each response to client.
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:16]: Thus, each Orion client needs a certificate/key pair to sign each query and transaction it submits to server, and the server has to have its own certificate/key pair to sign each response to client.

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 5]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+
+From another side, each Orion server has to keep each client certificate to validate client signatures and each client has to have access to server certificate to validate servers signatures. Both server and client need access to CA certificate to validate new client/server certificates.
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:18]: In order to validate client signatures, the Orion server has to keep the certificate of every client. This is achieved by the process of user on-boarding (add link to user management). Conversely, in order to validate the server's signature on responses, each client has to have access to server certificate. This is achieved by reading the ClusterConfig.Nodes (link to API). Both server and client need access to a CA certificate chain to validate client/server (resp.) certificates.

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 6]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+
+So, to start Orion new server, each server has to have following:
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:20]: start a new Orion server, each server has to have the following:

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 7]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+1. CA certificate (we support multiple CAs, including multilevel CAs).
+   * `caConfig.rootCACertsPath` 
+2. Its own certificate, sign by CA from above and corresponding private key.
+   * `server.identity.certificatePath` and `server.identity.keyPath`
+3. Default administrative user certificate, usually denoted as `admin` certificate.
+   * `admin.certificatePath`
+
+To work with Orion server, each client needs:
+1. CA certificate.
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:29]: Root CA certificate

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 8]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+### Crypto materials configuration
+
+Hyperledger Orion uses public key cryptography from user side, for user authentication and transaction Non-Repudiation. From server side, Orion uses PKI to sign and verify all server responses to clients. 
+
+Thus, each Orion client needs certificate/key pair to sign each query and transaction he submits to server, and server has to have its own certificate/key pair to sign each response to client.
+
+From another side, each Orion server has to keep each client certificate to validate client signatures and each client has to have access to server certificate to validate servers signatures. Both server and client need access to CA certificate to validate new client/server certificates.
+
+So, to start Orion new server, each server has to have following:
+1. CA certificate (we support multiple CAs, including multilevel CAs).
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:21]: Root CA

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 9]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+To work with Orion server, each client needs:
+1. CA certificate.
+```go 
+&config.ConnectionConfig{     
+   RootCAs: []string{
+      "./crypto/CA/CA.pem",
+   },
+}
+```
+2. Its own certificate, sign by CA from above and corresponding private key.
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:37]: the root CA

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 10]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+	UserConfig: &config.UserConfig{
+		UserID:         userID,
+		CertPath:       "./crypto/" + userID + "/" + userID + ".pem",
+		PrivateKeyPath: "./crypto/" + userID + "/" + userID + ".key",
+	},
+}
+```
+3. Server certificate can be accessed using node config query, [here](../queries/curl/node-config) and [here](../queries/gosdk/node-config).
+
+> For now, we ignore TLS configuration and location of TLS certificates, both servers and clients.
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:49]: the location
for both

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 11]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+
+### Crypto materials generation
+To generate new crypto materials, we use `scripts/cryptoGen.sh`. As first argument, it accepts absolute path to folder to store the generated files.
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:52]: The first argument is the absolute path to the folder used to store the generated files.

File: docs/getting-started/launching-one-node/crypto-materials.md [hunk 12]
```diff
@@ -9,3 +9,66 @@ title: Creating Crypto Material
...
+
+This run will generate 4 folders under `deployment/crypto`:
+1. CA
+2. server
+3. admin
+4. user
+
+If you need more certificates (for extra client or extra server nodes), they can be specified as extra arguments: 
+
+`./scripts/cryptoGen.sh deployment <extra users>`
```
Comment (tock-ibm) [inline docs/getting-started/launching-one-node/crypto-materials.md:71]: `./scripts/cryptoGen.sh deployment <extra users>`
for example
`./scripts/cryptoGen.sh deployment alice bob`

File: docs/getting-started/launching-one-node/crypto-materials.md (hunk 2 above)
Comment (gennadylaventman) [reply docs/getting-started/launching-one-node/crypto-materials.md:35]: yes, we use same `CACertCollection`as server.

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**